src/Tec/Plugin.php

```diff
@@ -104,26 +139,1020 @@ public function register() {
+				$msg .= $update_successful ? "successful" : "NOT successful";
+				$this->add_to_log( $msg );
+			}
+		}
+	}
+
+	/**
+	 * Re-save _tribe_default_ticket_provider for tribe_event.
+	 * WP All Import runs wp_unslash() and destroys "TEC\Tickets\Commerce\Module"
+	 *
+	 * @param array $record  The post data.
+	 * @param int   $post_id The new post ID.
+	 *
+	 * @return void
+	 */
+	public function resave_ticket_provider_for_event( array $record, int $post_id ): void {
+		if (
+			! empty( $record['_tribe_default_ticket_provider'] )
+		    && $record['_tribe_default_ticket_provider'] == "TEC\Tickets\Commerce\Module"
+		) {
+			if ( $this->fix_ticket_provider( $post_id ) ) {
+				$this->add_to_log( "Ticket provider successfully updated." );
+			} else {
+				$this->add_to_log( "Ticket provider update failed.");
+			}
+		}
+	}
+
+	/**
+	 * Update post_name (new id) and post_parent (new order id) for Tickets Commerce Attendees.
+	 *
+	 * @param array $record  The post data.
+	 * @param int   $post_id The new post ID.
+	 *
+	 * @return void
+	 */
+	public function maybe_update_post_data_for_attendee( array $record, int $post_id ): void {
+		$stop = false;
+
+		// If there is no record for the parent in the source data, then stop.
+		if ( ! isset( $record['parent'] ) ) {
+			$this->add_to_log( 'Post parent missing from import data...' );
+			$stop = true;
+		} else {
+			// Hash the old linked post type ID (tec_tc_order).
+			$this->meta_value = $this->hashit( $record['parent'] );
+			$this->meta_key   = "_tec_tc_order_export_hash";
+
+			// Grab the new post ID based on the hash.
+			$new_parent = $this->grab_post_id_based_on_meta();
+
+			// If the parent cannot be found in the database then stop.
+			if ( $new_parent == null ) {
+				$this->add_to_log( 'Post parent could not be found in database...' );
+				$stop = true;
+			}
+		}
+
+		if ( $stop ) {
+			$this->add_to_log( 'Deleting post' );
+			wp_delete_post( $post_id );
+		} else {
+			$args    = [
+				'ID'             => $post_id,
+				'post_name'      => $post_id,
+				'post_parent'    => $new_parent,
+				'comment_status' => 'closed',
+				'ping_status'    => 'closed',
+			];
+			$success = wp_update_post( $args );
+
+			// Logging
+			$msg = "Updating post name and post parent for Attendee ";
+			$msg .= $success ? "successful" : "NOT successful";
+			$this->add_to_log( $msg );
+		}
+	}
+
+	/**
+	 * Hashing function.
+	 *
+	 * @param string $subject The subject to be hashed.
+	 *
+	 * @return string The hashed string.
+	 */
+	function hashit( string $subject ): string {
+		return hash( 'sha256', $subject, false );
+	}
+
+	/**
+	 * Resets the delimiter to the pipe (|) character and creates an array of values.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param mixed $value The delimiter separated meta value.
+	 *
+	 * @return string[]    String or array of meta values.
+	 */
+	private function maybe_explode( mixed $value ): array {
```

Comment:
Type-hinting `mixed` requires PHP 8.0+, is this plugin requiring PHP 8.0+?

Reply:
Yes, it requires PHP 8.0+